common/lib/xmodule/xmodule/contentstore/mongo.py

```diff
@@ -406,7 +406,6 @@ def ensure_indexes(self):
                 ('_id.course', pymongo.ASCENDING),
                 ('_id.category', pymongo.ASCENDING)
             ],
-            sparse=True,
```

Comment:
No "sparse=True" because at least one of the keys above will be present in each document:
https://docs.mongodb.org/v3.0/core/index-sparse/#sparse-compound-indexes
AND - because the index is already created on the MongoDB instances without it.